### PR TITLE
AML - initial set of AML rules and severity update

### PR DESCRIPTION
**Pull request by batemansogq**
## PR Summary

the initial set of AML rules and severity update. 
This replaces an earlier pull request, which was extended. 

## PR Checklist

- [ ] PR has a meaningful title
- [ ] Summarized changes
- [ ] Change is not breaking
- [ ] This PR is ready to merge and is not **Work in Progress**
- **Rule changes**
  - [ ] Unit tests created/ updated
  - [ ] Rule documentation created/ updated
  - [ ] Link to a filed issue
  - [ ] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section
- **Other code changes**
  - [ ] Unit tests created/ updated
  - [ ] Link to a filed issue
  - [ ] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section

### CODE REVIEW

**Review by BernieWhite — CHANGES_REQUESTED** [review]
@batemansogq I've migrated most of the changes over from the other PR.

Please double check and make sure you're ok with my updates.

File: src/PSRule.Rules.Azure/rules/Azure.ML.Rule.yaml [hunk 1]
```diff
@@ -4,3 +4,130 @@
...
+    ruleSet: 2023_12
+    Azure.WAF/pillar: Security
+  labels:
+    Azure.MCSB.v1/control: [ 'NS-2' ]
+spec:
+  type:
+  - Microsoft.MachineLearningServices/workspaces
+  condition:
+    field: properties.publicNetworkAccess
+    equals: Disabled
```
[the remark this answers is not in the document]
Comment (batemansogq) [reply src/PSRule.Rules.Azure/rules/Azure.ML.Rule.yaml:108]: from the working sheet, these were intended as 2 separate rules; 

1. "Azure Machine Learning Workspaces should disable public network access" based upon https://portal.azure.com/#view/Microsoft_Azure_Policy/PolicyDetailBlade/definitionId/%2Fproviders%2FMicrosoft.Authorization%2FpolicyDefinitions%2F438c38d2-3772-465a-a9cc-7a6666a275ce
2. "Disable public network access when behind vnet" - from https://github.com/Azure/azure-synapse-analytics-end2end/blob/main/Deploy/modules/AIServicesDeploy.bicep it also has a policy ref = https://portal.azure.com/#view/Microsoft_Azure_Policy/PolicyDetailBlade/definitionId/%2Fproviders%2FMicrosoft.Authorization%2FpolicyDefinitions%2Fa10ee784-7409-4941-b091-663697637c0f

@nfmoore - can you please review and advise?

[the remark this answers is not in the document]
Comment (nfmoore) [reply src/PSRule.Rules.Azure/rules/Azure.ML.Rule.yaml:108]: According to the resource definition [here](https://learn.microsoft.com/en-us/azure/templates/microsoft.machinelearningservices/workspaces?pivots=deployment-language-bicep#workspaceproperties) the allowPublicAccessWhenBehindVnet and publicNetworkAccess properties are distinct with different functions. I think these should both remain.

[the remark this answers is not in the document]
Comment (BernieWhite) [reply src/PSRule.Rules.Azure/rules/Azure.ML.Rule.yaml:108]: @nfmoore Ok got it. So we should ask a customer to configure both.

[the remark this answers is not in the document]
Comment (BernieWhite) [reply src/PSRule.Rules.Azure/rules/Azure.ML.Rule.yaml:108]: Definitely this is going to be confusing for customers. For my own understanding what is the difference?

> There are two possible properties that you can configure:
>allow_public_access_when_behind_vnet - used by the Python SDK v1
>public_network_access - used by the CLI and Python SDK v2 Each property overrides the other. For example, setting >public_network_access will override any previous setting to allow_public_access_when_behind_vnet.
>Microsoft recommends using public_network_access to enable or disable public access to a workspace.

[the remark this answers is not in the document]
Comment (nfmoore) [reply src/PSRule.Rules.Azure/rules/Azure.ML.Rule.yaml:108]: @BernieWhite - you're right. with the v2 changes we should be using the public_network_access only.

[the remark this answers is not in the document]
Comment (BernieWhite) [reply src/PSRule.Rules.Azure/rules/Azure.ML.Rule.yaml:108]: @nfmoore Ok thanks for confirming that.

**Review by BernieWhite — APPROVED** [review]
Thanks @batemansogq.

Double check my changes. But if you're ok with them, we're all good to merge.

Let me know.

**Comment by nfmoore** [comment]
Looks good to me @BernieWhite. @batemansogq what are your thoughts ?

**Comment by batemansogq** [comment]
@BernieWhite & @nfmoore - happy roll with the changes. I can rebaseline and start looking at the other rules.

**Pull request merged**